webpack/assets/javascripts/react_app/Root/Context/ForemanContext.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+const ForemanContext = React.createContext({});
+
+export const ForemanProvider = ForemanContext.Provider;
+export const ForemanConsumer = ForemanContext.Consumer;
```

Comment:
Should we provide this at all and support it? or should we prefer the hook approach?

Reply:
@tbrisker If we only support hooks then you wouldn't be able to use ForemanContext in any existing class-based components.  React plans to continue supporting class components indefinitely, so I think it's a good idea to offer both options, instead of expecting people to go back and refactor a lot of old components.  (Also, Provider is still required even if the hook is used to consume it.)

Reply:
Not a strong feeling either way about this. My thinking is if needed a component could still import `ForemanContext` and use `ForemanContext.Consumer` directly, but if we are setting the "best practice" here of how to use the context, we don't have to give both options.

Reply:
Because most of our components are still classes with lifecycles, I thought it make sense to let the consumer choose between the two, but my vision is to adopt hooks as out best practice

Reply:
@amirfefer same, but also I think @tbrisker has a good point about 'nudging' consumers to choose hooks by making it a little less easy to do it the old way.